#156006054 Display collections to the user

muhallan opened the pull request:
#### What does this PR do?
Displays the collections that a user has created and also enables him/her to create a new one
#### Description of Task to be completed?
- Show new collections on the same page as the user adds them
- Attach a team to a collection
- Show each collection is a card format
- Ensure that each collection is clickable
- Implement the deleting of collections
#### How should this be manually tested?
Checkout to this branch and navigate to `/collections`
#### What are the relevant pivotal tracker stories?
#156006054
#### Screenshots (if appropriate)
Before deleting
![image](https://user-images.githubusercontent.com/27637970/37597286-10b4a342-2b90-11e8-8b24-839062b2419d.png)

After deleting some collections
![image](https://user-images.githubusercontent.com/27637970/37597339-3bf8989c-2b90-11e8-9fb2-d40ec043dd8e.png)

#### Questions:
Why doesn't delete have a confirm dialog?
- This proved to deserve a fresh independent task

muhallan committed:
feat(collections): display collections to the user

- show new collections on the same page as the user adds them
- show each collection is a card format
- ensure the collections are clickable
- implement the deleting of collections
[Finishes #156006054]

Ashaba reviewed (commented):
There are no tests for this.

Ashaba reviewed (commented):
No tests??

muhallan committed:
feat(collections): prevent duplicate collections

- stop adding collections that have similar names by the same user
- display an error message indicating the reason why one can't create the collection
